subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/DependencyManagementBuildScopeServices.java

```diff
@@ -747,25 +748,25 @@ ExecutionEngine createExecutionEngine(
             new IdentityCacheStep<>(
             new AssignWorkspaceStep<>(
             new LoadPreviousExecutionStateStep<>(
-            new CaptureStateBeforeExecutionStep(buildOperationExecutor, classLoaderHierarchyHasher, outputSnapshotter, overlappingOutputDetector,
+            new RemoveUntrackedExecutionStateStep<>(
+            new CaptureStateBeforeExecutionStep<>(buildOperationExecutor, classLoaderHierarchyHasher, outputSnapshotter, overlappingOutputDetector,
             new ValidateStep<>(virtualFileSystem, validationWarningRecorder,
-            new NoOpCachingStateStep(
+            new NoOpCachingStateStep<>(
             new ResolveChangesStep<>(changeDetector,
             new SkipUpToDateStep<>(
-            new BroadcastChangingOutputsStep<>(outputChangeListener,
             new StoreExecutionStateStep<>(
+            new BroadcastChangingOutputsStep<>(outputChangeListener,
```

Comment:
Why change the order of `BroadcastChangingOutputsStep` and `StoreExecutionStateStep`?

Reply:
Just an accident. But the order doesn't matter. One of the things I'd like to do eventually is to split steps into two, so that we have a pipeline that builds up context, and a pipeline that processes results returned. That would move these two steps to separate pipelines, removing the necessity to argue about their ordering.

I've reverted the change.